src/LuaPrinter.ts

```diff
@@ -31,7 +31,7 @@ export const escapeString = (value: string) => `"${value.replace(escapeStringReg
  * `foo.bar` => passes (`function foo.bar()` is valid)
  * `getFoo().bar` => fails (`function getFoo().bar()` would be illegal)
  */
-const isValidLuaFunctionDeclarationName = (str: string) => /^[a-zA-Z0-9_\.]+$/.test(str);
+const isValidLuaFunctionDeclarationName = (str: string) => /^[a-zA-Z0-9_.]+$/.test(str);
```

Comment:
Is this change correct? Seems unintended

Reply:
It was reported by [no-useless-escape](https://eslint.org/docs/rules/no-useless-escape) - the escape there is useless, because in square brackets all special characters (except closing bracket) don't have to be escaped